test/patch/requestAnimationFrame.spec.js

```diff
@@ -3,63 +3,82 @@
-      testZone.run(function() {
-        if (typeof mozRequestAnimationFrame === 'undefined') {
-          console.log('WARNING: skipping mozRequestAnimationFrame test (missing this API)');
-          return done();
-        }
-
-        // Some browsers (especially Safari) do not fire mozRequestAnimationFrame
-        // if they are offscreen. We can disable this test for those browsers and
-        // assume the patch works if setTimeout works, since they are mechanically
-        // the same
-        mozRequestAnimationFrame(function () {
-          expect(zone).toBeDirectChildOf(testZone);
+      it('should execute in the current zone if it is not the root', function (done) {
+        testZone.run(function () {
+          rAF(function () {
+            expect(zone === testZone).toBe(true);
+            done();
+          });
+        });
+      });
+      it('should fork a new zone if run from the root zone', function (done) {
+        expect(zone.isRootZone()).toBe(true);
+        rAF(function () {
+          expect(zone.parent.isRootZone()).toBe(true);
           done();
         });
       });
-    });
-  });
 
-  describe('webkitRequestAnimationFrame', function () {
-    it('should run the passed callback in a zone', function (done) {
-      testZone.run(function() {
-        if (typeof webkitRequestAnimationFrame === 'undefined') {
-          console.log('WARNING: skipping webkitRequestAnimationFrame test (missing this API)');
-          return done();
-        }
+      it('should bind to parent zone when called recursively', function (done) {
+        testZone.run(function () {
+          var frames = 0;
 
-        // Some browsers (especially Safari) do not fire webkitRequestAnimationFrame
-        // if they are offscreen. We can disable this test for those browsers and
-        // assume the patch works if setTimeout works, since they are mechanically
-        // the same
-        webkitRequestAnimationFrame(function () {
-          expect(zone).toBeDirectChildOf(testZone);
-          done();
+          function frameCallback() {
+            expect(zone === testZone).toBe(true);
+            if (frames++ > 15) {
+              return done();
+            }
+            rAF(frameCallback);
+          }
+
+          rAF(frameCallback);
         });
       });
+
+      //it('should bind to parent zone when called recursively in an async fashion', function (done) {
+      //  testZone.run(function () {
+      //    var frames = 0;
+      //
+      //    function frameCallback() {
+      //      // For this check, since we're injecting ANOTHER zone by using
+      //      // setTimeout, we need to expect that the parent zone will be
+      //      // our test zone.
+      //      var expectedZone = frames ? zone.parent : zone;
+      //      expect(expectedZone === testZone).toBe(true);
+      //      if (frames++ > 15) {
+      //        return done();
+      //      }
+      //      setTimeout(function () {
```

Comment:
> Sadly, I can easily see someone doing this. Imagine you want to use requestAnimationFrame but only periodically, so you call setTimeout in your handler and then request the next frame when the handler is triggered. Sure this would be misguided, but hopefully zone.js wouldn't fail to work seamlessly because you've constructed some questionable code.

Unless you have a concrete use case in mind, I would revert to the simpler solution. IMO when rAF gets used, you are interested in getting called for each single frame.